server/src/main/java/com/ncc/neon/controllers/MutateController.java

```diff
@@ -68,4 +62,38 @@
 
         return ResponseEntity.ok().body(queryService.mutateData(info, mutateQuery));
     }
+
+    @PostMapping(path="/byfilter", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<Mono<ActionResult>> mutateDataByFilter(@RequestBody MutateQuery mutateQuery) {
+        final Predicate<String> isBlank = StringUtils::isBlank;
+        final Predicate<Collection> isEmpty = CollectionUtils::isEmpty;
+
+        Triple[] labeledInput = new Triple[]{
+                Triple.of("Datastore Host", mutateQuery.getDatastoreHost(), isBlank),
+                Triple.of("Datastore Type", mutateQuery.getDatastoreType(), isBlank),
+                Triple.of("Database Name", mutateQuery.getDatabaseName(), isBlank),
+                Triple.of("Table Name", mutateQuery.getTableName(), isBlank),
+                Triple.of("Fields with Values", mutateQuery.getFieldsWithValues().entrySet(), isEmpty)
+        };
+
+        List<String> invalidInput = Arrays.asList(labeledInput).stream()
+                .filter(triple -> ((Predicate)triple.getRight()).test(triple.getMiddle()))
+                .map(triple -> (String)triple.getLeft()).collect(Collectors.toList());
+
+        if (mutateQuery.getWhereClause() == null) {
+            invalidInput.add("Where Clause");
+        }
+
+        if (invalidInput.size() > 0) {
+            return ResponseEntity.badRequest().body(Mono.just(new ActionResult("Mutation by ID Query Missing " +
```

Comment:
Change string to "Mutation by Filter Query Missing"

Reply:
Fixed.